project-base/storefront/components/Pages/Order/PaymentConfirmation/Gateways/GoPayGateway.tsx

```diff
@@ -1,21 +1,29 @@
-import { Link } from 'components/Basic/Link/Link';
+import { SpinnerIcon } from 'components/Basic/Icon/IconsSvg';
+import { Button } from 'components/Forms/Button/Button';
 import { GoPayCreatePaymentSetupApi, usePayOrderMutationApi } from 'graphql/generated';
 import { showErrorMessage } from 'helpers/toasts';
 import useTranslation from 'next-translate/useTranslation';
 import Script from 'next/script';
-import { useEffect, useState } from 'react';
+import { useEffect, useRef, useState } from 'react';
 
 type GoPayGatewayProps = {
     orderUuid: string;
+    requiresAction?: boolean;
+    initialButtonText?: string;
 };
 
-export const GoPayGateway: FC<GoPayGatewayProps> = ({ orderUuid }) => {
-    const { t } = useTranslation();
-    const [isRedirectLinkVisible, setRedirectLinkVisibility] = useState(false);
+export const GoPayGateway: FC<GoPayGatewayProps> = ({ orderUuid, requiresAction, initialButtonText, className }) => {
+    const [initiatedPaymentGate, setInitiatedPaymentGate] = useState(!requiresAction);
     const [goPayPaymentSetup, setGoPayPaymentSetup] = useState<GoPayCreatePaymentSetupApi | undefined>(undefined);
     const [, payOrder] = usePayOrderMutationApi();
+    const { t } = useTranslation();
+    const wasPaidRef = useRef(false);
 
     useEffect(() => {
+        if (wasPaidRef.current || !initiatedPaymentGate) {
+            return;
+        }
```

Comment:
As I've already explain to you somewhere else this way of handling `useEffect` function is good practice in my opinion. Since useEffect is not classic function where you would use this pattern it should only return cleanup function. If you can wrap the whole code into the condition why not to wrap it? I will leave it on your choice.